src/functions.rs

```diff
@@ -232,22 +233,13 @@ pub fn get_by_index(value: &[u8], index: usize) -> Option<Vec<u8>> {
     let header = read_u32(value, 0).unwrap();
     match header & CONTAINER_HEADER_TYPE_MASK {
         ARRAY_CONTAINER_TAG => {
-            let length = (header & CONTAINER_HEADER_LEN_MASK) as usize;
-            if index >= length {
-                return None;
-            }
-            let mut jentry_offset = 4;
-            let mut val_offset = 4 * length + 4;
-            for i in 0..length {
+            let offsets = get_offsets_by_index(value, 0, header, index);
```

Comment:
It looks like we could just return `jentry` instead of `jentry_offset`, so we can avoid duplicating decode